### PR TITLE
new waiting flag for buttons. plumb that through for tracker windows

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers

### CODE REVIEW

File: shared/actions/tracker.js [hunk 1]
```diff
@@ -275,14 +275,27 @@ function trackUser (trackToken: ?string): Promise<boolean> {
   })
 }
 
+function onWaiting (username: string, waiting: bool): (dispatch: Dispatch) => void {
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: we can follow this simple pattern elsewhere

File: shared/common-adapters/button.desktop.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-import {FlatButton} from 'material-ui'
+import {FlatButton, CircularProgress} from 'material-ui'
 import {globalStyles, globalColors, globalColorsDZ2} from '../styles/style-guide'
 import type {Props} from './button'
 
 export default class Button extends Component {
   props: Props;
 
-  render1 () {
+  _render1 () {
```
Comment (chrisnojima) [inline shared/common-adapters/button.desktop.js]: lets mark private members with _

File: shared/common-adapters/button.desktop.js [hunk 3]
```diff
@@ -94,29 +98,53 @@ export default class Button extends Component {
...
-      <FlatButton
-        onClick={this.props.onClick}
-        style={{...backgroundStyle, ...smallStyle, ...this.props.style}}
-        labelStyle={{...styles2.buttonLabel, ...labelStyle}}
-        label={this.props.label || this.props.more && '•••'}
-        primary={this.props.type === 'Primary'}
-        secondary={this.props.type === 'Secondary'}
-        disabled={this.props.disabled}
-      />
+      <div style={outterStyle}>
```
Comment (chrisnojima) [inline shared/common-adapters/button.desktop.js]: we wrap the button. the button itself gets opacity changes when disabled so let's not have our overlay get that too

File: shared/common-adapters/button.desktop.js [hunk 4]
```diff
@@ -136,6 +164,7 @@ const buttonCommon1 = {
 const buttonCommon2 = {
   ...globalStyles.DZ2.fontSemibold,
   color: globalColorsDZ2.white,
+  whiteSpace: 'nowrap',
```
Comment (chrisnojima) [inline shared/common-adapters/button.desktop.js]: let's disallow our buttons wrapping and being all messed up. you can see this on master if you squish the view

File: shared/more/dev-container.desktop.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import {Paper, AppBar} from 'material-ui'
 export default class Container extends Component {
   render () {
     return (
-      <Paper zDepth={5} style={{margin: 20}}>
+      <Paper zDepth={5} style={{...this.props.style, margin: 20}}>
```
Comment (chrisnojima) [inline shared/more/dev-container.desktop.js]: this was never plumbed through

File: shared/more/style-sheet.render.desktop.js [hunk 6]
```diff
@@ -71,7 +71,7 @@ export default class Render extends Component {
           )}
           </div>
         </Container>
-        <Container title='Buttons'>
+        <Container title='Buttons' style={{order: -1}}>
```
Comment (chrisnojima) [inline shared/more/style-sheet.render.desktop.js]: little trick to bring this to the top the help debugging

File: shared/more/style-sheet.render.desktop.js [hunk 7]
```diff
@@ -90,13 +90,26 @@ export default class Render extends Component {
                 <Space/>
                 <Button dz2 onClick={() => {}} type='Follow' disabled label='Follow disabled'/><Space/>
               </div>
+              <div style={{...globalStyles.flexBoxColumn, alignItems: 'flex-start', justifyContents: 'flex-start', padding: 10, paddingRight: 100}}>
```
Comment (chrisnojima) [inline shared/more/style-sheet.render.desktop.js]: new versions w/ wait on

File: shared/common-adapters/button.desktop.js [hunk 8]
```diff
@@ -94,29 +98,53 @@ export default class Button extends Component {
...
     }
 
+    if (this.props.waiting) {
+      labelStyle = {
+        ...labelStyle,
+        opacity: 0
+      }
+    }
+
+    let outterStyle = {position: 'relative'}
```
Comment (MarcoPolo) [inline shared/common-adapters/button.desktop.js]: typo? should be `outer`

**Comment by MarcoPolo** [comment]
can we use black75 for the spinner on secondary buttons?

It's invisible otherwise:
screen shot 2016-03-03 at 10 44 37 am
screen shot 2016-03-03 at 10 29 24 am

**Comment by MarcoPolo** [comment]
:+1:

**Pull request merged**